File: 70.爬楼梯.py
#
# @lc app=leetcode.cn id=70 lang=python3
#
# [70] 爬楼梯
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        f = [1,2]
        if n>=3:
            for i in range(2,n):
                tmp = f[i-1] + f[i-2]
                f.append(tmp)
            return f[-1]
        else:
            return f[n-1]
# @lc code=end

File: 70.爬楼梯.java
/*
 * @lc app=leetcode.cn id=70 lang=java
 *
 * [70] 爬楼梯
 */

// @lc code=start
class Solution {
    public int climbStairs(int n) {
        int[] s = new int[n+1];
        s[0] = 1;
        s[1] = 1;
        if (n < 2){
            return 1;
        }else{
            for(int i=2;i<=n;i++) s[i] = s[i-1] + s[i-2];
        }

        return s[n];
    }
}
// @lc code=end

